Tests/Client/Custom Assertions/AssertNetworkRequest.swift

```diff
@@ -9,112 +9,86 @@ import XCTest
-                          path: String,
-                          headers: [String: String]?,
-                          queryParameters: [String: String]?,
-                          body: [String: Any]?,
-                          timeout: TimeInterval = 0.5,
-                          file: StaticString = #file,
-                          line: UInt = #line) {
-    
-    guard let request = RequestRecorderURLProtocol.waitForRequest(timeout: timeout) else {
-        XCTFail("Waiting for request timed out. No request was made.", file: file, line: line)
-        return
-    }
-    
-    var errorMessage = ""
-    defer {
-        if !errorMessage.isEmpty {
-            XCTFail("AssertNetworkRequest failed:" + errorMessage, file: file, line: line)
-        }
-    }
-    
-    // Check method
-    if method.rawValue != request.httpMethod {
-        errorMessage += "\n  - Incorrect method: expected \"\(method.rawValue)\" got \"\(request.httpMethod ?? "_")\""
-    }
-    
-    guard let url = request.url, let components = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
-        errorMessage += "\n  - Missing URL"
-        return
-    }
-    
-    // Check path
-    if components.path != path {
-        errorMessage += "\n  - Incorrect path: expected \"\(path)\" got \"\(components.path)\""
+extension Assert {
+    /// Synchronously waits for a network request that matches its properties with the given parameters.
+    ///
+    /// The function periodically checks the `RequestRecorderURLProtocol.recordedRequests` and If no request matches the
+    /// given parameters within the `timeout` period, this assertion fails with the time-out error.
+    ///
+    /// The values specified in the `headers`, `queryParameters` and `body` represents the mandatory subset of
+    /// the values the request must have. A request is valid even when it contains additional parameters than
+    /// the ones specified in these values.
+    ///
+    /// - Parameters:
+    ///   - method: The HTTP method the request.
+    ///   - path: The `path` part of the request's URL.
+    ///   - headers: The headers required for the request.
+    ///   - queryParameters: The query parameters required for the request.
+    ///   - body: The expected body of the request.
+    ///   - timeout: The maximum time this function waits for a request to match the given parameters.
+    ///
+    static func networkRequest(method: Endpoint.Method,
+                               path: String,
+                               headers: [String: String]?,
+                               queryParameters: [String: String]?,
+                               body: [String: Any]?,
+                               timeout: TimeInterval = 0.5,
+                               file: StaticString = #file,
+                               line: UInt = #line) -> Assertion {
+        
+        return Assert.willBeTrue(RequestRecorderURLProtocol.recordedRequests.contains {
```

Comment:
Did you test how this behaves when the test fails? I think you need to pass the `file` and `line` parameters to this assertion, too, otherwise Xcode will show the error on this line, which is not really helpful.